### ISSUE TITLE
dimension ordering in climt

### ISSUE THREAD

**Issue by JoyMonteiro** [CliMT/climt#77]
I don't remember where we were discussing this, but we had decided to keep the dimension ordering to suit the dynamical core (i.,e horizontal dimensions are the fastest varying indices).

On second thought, I don't think this is the correct decision. In an MPI scenario, this would mean that 
there is communication needed to run components which require a whole column. Since the dycore will
anyway require communications (since it is not scalable to distribute arrays based on vertical levels),
this would put a huge overhead.

Even in our current scenario, this would mean array reshaping for every component other the GFS, 
which is again inefficient.

I propose we move back to the original ordering. I can do this. I presume all this would entail is changing
the ordering in the properties those components which are lat/lon aware?

**Comment by mcgibbon** [comment]
> On second thought, I don't think this is the correct decision. In an MPI scenario, this would mean that
there is communication needed to run components which require a whole column.

I don't think this is true. The processors can still be split along latitude/longitude indices, even if those are the fastest-varying indices.

> Even in our current scenario, this would mean array reshaping for every component other the GFS,
which is again inefficient.

This is definitely not true. For the Fortran-wrapped components, the Fortran drivers pretty much all (or actually all) contained ways to specify number of latitudes and longitudes, so that they could operate on arrays whose slowest-varying index is the vertical (what the GFS core wants). Currently there is no array reshaping that occurs for those components.

With the pure Python components, they also generally won't require array reshaping because we've written them to use the same ordering of dims that GFS uses. However, even if they were to use reshaping, for something like a transpose there is no data copying that takes place. The pure Python components will take advantage of numpy array strides, so no copying is necessary to change the dimension ordering in this way.

I propose we keep the current ordering unless we're absolutely sure we need to change it, which I'm not convinced of, based on the above.

**Comment by JoyMonteiro** [comment]
Yes, I see in the code what you mean by rrtmg works with the vertical in the slowest index.

I'm not sure about the MPI either, so I'll close this issue and we can discuss this later  if needed when MPI based
components become a reality.
